create_key/create_key.go

```diff
@@ -0,0 +1,121 @@
+package createKey
+
+import (
+	"errors"
+	"flag"
+	"fmt"
+	"os"
+	"path/filepath"
+	"strings"
+
+	"github.com/NBISweden/sda-cli/helpers"
+	"github.com/elixir-oslo/crypt4gh/keys"
+	log "github.com/sirupsen/logrus"
+)
+
+// Help text and command line flags.
+
+// Usage text that will be displayed as command line help text when using the
+// `help encrypt` command
+var Usage = `
+USAGE: %s createKey <name> (-outdir <dirname>)
+
+createKey: Creates a crypt4gh encryption key pair, and saves it to
+           <name>.pub.pem, and <name>.sec.pem.
+           NOTE: keys created using this function should not be used when
+           encrypting submission files, they should only be used for decrypting
+           files downloaded from the archive.
+`
+
+// ArgHelp is the suffix text that will be displayed after the argument list in
+// the module help.
+var ArgHelp = ``
+
+// Args is a flagset that needs to be exported so that it can be written to the
+// main program help
+var Args = flag.NewFlagSet("createKey", flag.ExitOnError)
+
+var outDir = Args.String("outdir", "", "Output directory for the key files")
+
+// CreateKey takes two arguments, a base filename, and optionally an output
+// directory specified with `-outdir`.
+func CreateKey(args []string) error {
+
+	// Parse flags. There are no flags at the moment, but in case some are added
+	// we check for them.
+	err := Args.Parse(args[1:])
+	if err != nil {
+		return fmt.Errorf("could not parse arguments: %s", err)
+	}
+
+	// Args() returns the non-flag arguments, which we assume is the key
+	// filename. If more than one name is given, an error is returned.
+	if len(Args.Args()) > 1 {
+		return fmt.Errorf("unknown arguments: %v, expected a single filename", strings.Join(Args.Args(), ", "))
+	}
+	if len(Args.Args()) < 1 {
+		return errors.New("no filename given")
+	}
+	basename := Args.Args()[0]
+
+	// Add the output directory to the file path (does nothing if outDir is "")
+	basename = filepath.Join(*outDir, basename)
+
+	// Write the key files
+	err = generateKeyPair(basename, "")
```

Comment:
No password for the secret key?
How do you really think this tool is going to be used? entirely in a scripted environment?

Reply:
well - my goal was to make it as simple as possible, and since a lot of people won't give a meaningful password anyway, I figured we could skip it :) do you think we should enforce a password instead?

Reply:
This app is what we say our users should use, then we should actively promote high security standards.
And since key creation is not something that will be scripted in most cases I say we enforce the use of passphrase protection.

Reply:
great! :) I'll make it passwordy!